ksql-common/src/main/java/io/confluent/ksql/function/UdfFactory.java

```diff
@@ -74,6 +74,10 @@ public String toString() {
   }
 
   public KsqlFunction getFunction(final List<Schema.Type> paramTypes) {
-    return functions.get(paramTypes);
+    final KsqlFunction function = functions.get(paramTypes);
+    if (function == null) {
+      throw new KsqlException("Function '" + name + "' does not accept parameters of types:" + paramTypes);
+    }
+    return function;
```

Comment:
For the error message lets' use the type names consistent with the types user sees. You can use `TYPE_MAP` in `SchemaUtil.java`. The type names you use here are internal type names.

Reply:
Cool. Didn't know about that.

Reply:
Couple of questions about `TYPE_MAP` and `SchemaUtil`:
1. There doesn't seem to be an entry in this  `TYPE_MAP` for `STRUCT`... should there be, or is that coming in another PR?
1. There are several methods / structures with `SchemaUtil` that contain SQL type names, e.g. `getTypeSchema, `TYPE_MAP`, `getSqlTypeName`, but the names they use aren't consistent, e.g. `INT` vs `INTEGER`, some have `FLOAT32` / some don't, `VARCHAR(STRING)` vs `VARCHAR` - are these intentional or not?

Reply:
@big-andy-coates here are the answers for your questions:
1. https://github.com/confluentinc/ksql/pull/1401 adds STRUTC, along with MAP and ARRAY to the TYPE_MAP.
2. We may have some regression in SchemaUtil. For some types we intentionally decided to  accept multiple names. For instance, VARCHAR and STRING are valid names for string type. For any interaction with user we should use KSQL types and we should try to keep them as close to standard SQL types. 
In case of using `VARCHAR(STRING)` I think we only use it in describing a schema and not as a type name.
If you find any inconsistencies that please go ahead an fix them :) If you think we should make any changes in the KSQL type names let's discuss it more.